src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,369 @@
+
+    while (FindStreamTransportConnection(nextConnectionID) != false)
+    {
+        if (nextConnectionID == MAX_PUSH_TRANSPORT_CONNECTION_ID)
+            nextConnectionID = 0;
+        else
+            nextConnectionID = nextConnectionID + 1;
+    }
+    return nextConnectionID;
+}
+
+void PushAvStreamTransportServer::HandleAllocatePushTransport(HandlerContext & ctx,
+                                                              const Commands::AllocatePushTransport::DecodableType & commandData)
+{
+    Status status = Status::Success;
+    Commands::AllocatePushTransportResponse::Type response;
+    auto & transportOptions = commandData.transportOptions;
+
+    uint16_t connectionID                 = GenerateConnectionID();
+    TransportStatusEnum outTranportStatus = TransportStatusEnum::kUnknownEnumValue;
+
+    // call the delegate
+    status = mDelegate.AllocatePushTransport(connectionID, transportOptions, outTranportStatus);
+
+    if (status == Status::Success)
+    {
+        response.connectionID     = connectionID;
+        response.transportOptions = transportOptions;
+        response.transportStatus  = outTranportStatus;
+
+        // add connection to CurrentConnections
+        AddStreamTransportConnection(connectionID);
+
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+    }
+    else
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+    }
+}
+
+void PushAvStreamTransportServer::HandleDeallocatePushTransport(
+    HandlerContext & ctx, const Commands::DeallocatePushTransport::DecodableType & commandData)
+{
+    Status status         = Status::Success;
+    uint16_t connectionID = commandData.connectionID;
+
+    // Call the delegate
+    status = mDelegate.DeallocatePushTransport(connectionID);
+
+    if (status == Status::Success)
+        // Remove connection form CurrentConnections
+        RemoveStreamTransportConnection(connectionID);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void PushAvStreamTransportServer::HandleModifyPushTransport(HandlerContext & ctx,
+                                                            const Commands::ModifyPushTransport::DecodableType & commandData)
+{
+    Status status              = Status::Success;
+    uint16_t connectionID      = commandData.connectionID;
+    auto & outTransportOptions = commandData.transportOptions;
+
+    // Call the delegate
+    status = mDelegate.ModifyPushTransport(connectionID, outTransportOptions);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void PushAvStreamTransportServer::HandleSetTransportStatus(HandlerContext & ctx,
+                                                           const Commands::SetTransportStatus::DecodableType & commandData)
+{
+    Status status           = Status::Success;
+    uint16_t connectionID   = commandData.connectionID;
+    auto & transportOptions = commandData.transportOptions;
```

Comment:
This command does not have a TransportOptions field.

Was this even compiled?  I am guessing no....